src/hostcalls.rs

```diff
@@ -416,7 +416,7 @@ pub fn get_property(path: Vec<&str>) -> Result<Option<Bytes>, Status> {
                 }
             }
             Status::NotFound => Ok(None),
-            status => panic!("unexpected status: {}", status as u32),
+            status => Err(status),
```

Comment:
This catch-all panic for unexpected value is intentional in order to catch issues in host and/or SDK implementations, and it was invaluable in finding bugs in the past, so it should not be removed.

If there is a status value that is expected to work and it's missing, then it should be added explicitly.

Reply:
_I think_ you are proposing to add explicit branches to the `match`, along side the existing `Ok` and `NotFound`, right? What would those be tho? I know `SerializationFailure` is definitely one of them, but I don't know what a host is allowed to return here? Is that spec'ed? 

But mostly confused about what you are trying to reach as a goal here? `panic` is a harsh fallback... Nothing the module can do about... Should we maybe bake the assumptions about what `Status` are expected in the default implementation for the `pub trait Context` ? So if anything unexpected happens, it still gets raised (as today it'd panic at `get_property(path).unwrap()`) while leaving a chance to the user to mitigate the issue?

"Happy path" remains `Ok` & `NotFound`, that seems obvious. `ParseFailure`, `InvalidMemoryAccess`, `CasMismatch`, `ResultMismatch` seem to be off the table at first glance... which leaves the others as possible outcomes?

Reply:
> _I think_ you are proposing to add explicit branches to the `match`, along side the existing `Ok` and `NotFound`, right? What would those be tho? I know `SerializationFailure` is definitely one of them, but I don't know what a host is allowed to return here? Is that spec'ed?

Looking at Envoy's code, the missing values are `InternalFailure` and `SerializationFailure`.

The return values are generally spec'ed out (see: https://github.com/proxy-wasm/spec/pull/42), but the properties in the current form are a bit hand-wavy, so we're using Envoy's code as the source of truth for v0.2.1.

> But mostly confused about what you are trying to reach as a goal here? `panic` is a harsh fallback... Nothing the module can do about...

The `panic` here is meant to handle the case when the host returned unexpected (invalid) status code.

It's not about handling the "sad path".

Reply:
Changed as requested.